libraries/SITL/SIM_GPS.cpp

```diff
@@ -1171,4 +1160,40 @@ void GPS::update()
     }
 }
 
+/*
+  get delayed data by interpolation
+*/
+GPS::gps_data GPS::interpolate_data(const gps_data &d, uint32_t delay_ms)
+{
+    const uint8_t N = ARRAY_SIZE(_gps_history);
+    const uint32_t now_ms = d.timestamp_ms;
+
+    // add in into history array, shifting old elements
+    memmove(&_gps_history[1], &_gps_history[0], sizeof(_gps_history[0])*(ARRAY_SIZE(_gps_history)-1));
```

Comment:
Why isn't this a circular/ring buffer?

Reply:
> Why isn't this a circular/ring buffer?

because this is simpler, and simplicity beats cpu efficiency in SITL. memmove in SITL operates at gbyte/sec anyway